app/src/common/shared/com/igalia/wolvic/ui/widgets/KeyboardWidget.java

```diff
@@ -566,7 +566,8 @@ public void onConfigurationChanged(Configuration newConfig) {
     @Override
     public void onPress(int primaryCode) {
         Log.d(LOGTAG, "Keyboard onPress " + primaryCode);
-        mWidgetManager.triggerHapticFeedback();
+        if (mControlHoverDeviceId != -1)
```

Comment:
Can it be -1 ?

Reply:
Umm, if I've understood it correctly, [updateLastDeviceId](https://github.com/Igalia/wolvic/blob/04131b06656a49f871d85d5855d01c0e123e9562/app/src/common/shared/com/igalia/wolvic/ui/widgets/KeyboardWidget.java#L165) can set the deviceId to -1 in some case.

Reply:
So the part I don't get is why you check the control hover device id. That refers to the control buttons of the keyboard that are shown on the left (closing, moving, etc...)

Reply:
ah, I know what you mean. I should have used mPopUpHoverDeviceId instead.

Reply:
Why? I don't see any difference. I just don't understand why you're using those. They track whenever the hover is over any of those elements of the keyboard